activerecord/lib/active_record/relation.rb

```diff
@@ -298,6 +298,35 @@ def many?
       limit_value ? to_a.many? : size > 1
     end
 
+    # Returns a cache key that can be used to identify the records fetched by
+    # this query. The cache key is built with a fingerprint of the sql query,
+    # the number of records matched by the query and a timestamp of the last
+    # updated record. When a new record comes to match the query, or any of
+    # the existing records is updated or deleted, the cache key changes.
+    #
+    #   Product.where("name like ?", "%Cosmic Encounter%").cache_key
+    #   => "products/query-1850ab3d302391b85b8693e941286659-1-20150714212553907087000"
+    #
+    # Under the hood this triggers two SQL queries:
+    #
+    #    SELECT COUNT(*) FROM "products" WHERE (name like '%Cosmic Encounter%')
+    #    SELECT MAX("products"."updated_at") FROM "products" WHERE (name like '%Cosmic Encounter%')
+    #
+    # You can also pass a custom timestamp column to fetch the timestamp of the
+    # last updated record.
+    #
+    #   Product.where("name like ?", "%Game%").cache_key(:last_reviewed_at)
+    def cache_key(timestamp_column = :updated_at)
```

Comment:
Name is a bit misleading, it may be how the reason you *want* to use this data, but not what the data actually *is* (e.g. it doesn't refer to any actual cache stored by Rails). Not really sure what would be a better name, though...